### PR TITLE
{Network} `az network application-gateway identity/probe/redirect-config`: Migrate commands to codegen v2

### PR DESCRIPTION
**Pull request by necusjz**
**Related command**

az network application-gateway identity
az network application-gateway probe
az network application-gateway redirect-config

**Description**

Total: 16 commands

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -1318,41 +1325,33 @@ def post_instance_update(self, instance):
...
-                                     include_query_string=None, raw=False):
-    SubResource = cmd.get_models('SubResource')
-    if redirect_type:
-        instance.redirect_type = redirect_type
-    if target_listener:
-        instance.target_listener = SubResource(id=target_listener)
-        instance.target_url = None
-    if target_url:
-        instance.target_listener = None
-        instance.target_url = target_url
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:1350]: It seems need to clean up target_url if target_listener is not None and vice versa.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -1506,67 +1505,59 @@ def delete_ag_rewrite_rule_condition(cmd, resource_group_name, application_gatew
...
+        args_schema.protocol._required = True
+        args_schema.match._registered = False
+        return args_schema
 
+    def pre_operations(self):
+        args = self.ctx.args
+        if has_value(args.match_body):
+            args.match.body = args.match_body
+        if has_value(args.match_status_codes):
+            args.match.status_codes = args.match_status_codes
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:1531]: Why not flatten match?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 3]
```diff
@@ -1506,67 +1505,59 @@ def delete_ag_rewrite_rule_condition(cmd, resource_group_name, application_gatew
...
+        )
+        args_schema.match._registered = False
+        return args_schema
+
+    def pre_operations(self):
+        args = self.ctx.args
+        if has_value(args.match_body):
+            args.match.body = args.match_body
+        if has_value(args.match_status_codes):
+            args.match.status_codes = args.match_status_codes
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:1560]: Why not flatten match?

File: src/azure-cli/azure/cli/command_modules/network/custom.py (hunk 1 above)
Comment (necusjz) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:1350]: nice catch!

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**